### PR TITLE
feat: migrate BQ Java samples from java-docs-samples to client lib

### PR DESCRIPTION
**Pull request by stephaniewang526**
Fixes b/146052864

### CODE REVIEW

File: samples/pom.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    Removing or replacing it should not affect the execution of the samples in anyway.
+  -->
+  <parent>
+    <groupId>com.google.cloud.samples</groupId>
+    <artifactId>shared-configuration</artifactId>
+    <version>1.0.11</version>
+  </parent>
+
+  <properties>
+    <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (chingor13) [inline samples/pom.xml]: If we're compiling for Java 8, we need to skip this entire module for the Java 7 CI. We can do that with a profile in the root project where we conditionally change the list of modules.

File: samples/pom.xml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+
+  <dependencies>
+    <!-- [START bigquery_java_dependencies] -->
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-bigquery</artifactId>
+      <version>1.100.0</version>
```
Comment (chingor13) [inline samples/pom.xml]: This is an interesting problem to solve - we want to test against HEAD (usually the `-SNAPSHOT` version), but for the sample, we want the released version.

This can be solved in a separate PR.

File: samples/pom.xml (hunk 1 above)
Comment (stephaniewang526) [reply samples/pom.xml]: Updated PR

File: samples/pom.xml (hunk 2 above)
Comment (stephaniewang526) [reply samples/pom.xml]: I will keep the released version for now since this is the safest thing. We will need to look into incorporating release process into samples release.

**Comment by BenWhitehead** [comment]
We also need to disable release for the sample module. @chingor13 How do we do that with the current setup of things.
